### ISSUE TITLE
Notification in app that collected IDs will be lost and can't be transfered

### ISSUE THREAD

**Issue by TR0LLIP0P** [corona-warn-app/cwa-wishlist#285]
## Feature description
There should be a clear notification/note in the app (NOT the online documentation!) that the collected IDs will be lost if the app is removed and that you cannot transfer the collected IDs to a new smartphone.
There should be an instruction what to do if you get a new phone (have the old phone active for two weeks, ...).

## Problem and motivation
Christmas is two weeks away. I guess that lots of people will get new smartphones as a present. So they will transfer their personal data (google and apple make this possible with cloud backup) and use the new phone, not the old one.
I suspect that most of them do not know that the collected IDs of CWA will NOT be transferred. From this follows that much data collected in the two weeks before Christmas is useless.
A note in the online documentation is not enough, because most people don't look there.

## Is this something you're interested in working on
No

**Comment by MikeMcC399** [comment]
@TR0LLIP0P 
There is a related discussion in https://github.com/corona-warn-app/cwa-website/issues/411 about adding instructions to the FAQs concerning what to do when moving to a new mobile phone.

Again, this is only the FAQs not the in-app documentation, which you are targeting, but 
- https://www.coronawarn.app/en/faq/#backup covers the fact that backup doesn't include the CWA keys
- https://www.coronawarn.app/en/faq/#delete_random_ids discusses about whether keys are preserved on uninstall or not (for iOS: yes, for Android: no)

**Comment by achisto** [comment]
I would strongly encourage the developers to think about a one-time push notification explaining what @TR0LLIP0P pointed out and linking that notification to either the FAQ-page or an in-app explanation. I'm personally informed about two people who switched phones and haven't thought about this over the past weeks, I believe this might become a huge problem during Christmas.

**Comment by TR0LLIP0P** [comment]
I think most new phones are in use now. Christmas is over and sadly no notification appeared. So the main reason for my issue (christmas ->lots of new phones -> lots of keys useless) is gone.
I think I can close this issue?

**Comment by Ein-Tim** [comment]
@TR0LLIP0P 

I would leave this Issue open since a notification regarding this would be important regardless of Christmas...

**Comment by dsarkar** [comment]
Hi @TR0LLIP0P, @Ein-Tim, @achisto 

Thanks again for contributing. We will bring this issue again to the developers. 

From what I can see though, it is not possible for iOS devices to send a push notification on an uninstall event, see
https://developer.apple.com/forums/thread/116445 (last post).

DS

**Comment by TR0LLIP0P** [comment]
@dsarkar Why do you want to send a notification while uninstalling? If you uninstall the app it's too late. The reason for this issue was that people (probably) don't know that keys can't be transferred if they use a new phone. I don't think that people delete all apps from their old phone when they use a new one.

**Comment by TR0LLIP0P** [comment]
> @TR0LLIP0P
> 
> I would leave this Issue open since a notification regarding this would be important regardless of Christmas...

The problem is known since the beginning of the app and didn't seem important.
The main event where a lot of people use new phones (= lots of data is lost) is christmas.
So I think it's too late and resources can better be used for other issues.

**Comment by Ein-Tim** [comment]
@TR0LLIP0P 
But the question is: When should the user get informed that the data is not included in the backup? At the frost start of the App? At a random time? 
At the first start of the App seems best, but I think normal users will quickly forget about this...

If you want to close this, close it. We could continue the discussion in #8

**Comment by achisto** [comment]
> Why do you want to send a notification while uninstalling? If you uninstall the app it's too late. The reason for this issue was that people (probably) don't know that keys can't be transferred if they use a new phone. I don't think that people delete all apps from their old phone when they use a new one.
> The problem is known since the beginning of the app and didn't seem important.
> The main event where a lot of people use new phones (= lots of data is lost) is christmas.
> So I think it's too late and resources can better be used for other issues.

You are right, I believe that's what OP meant. Sending a notification on uninstall is not possible - although at least on iOS there is a notification shown that looks [like this](https://user-images.githubusercontent.com/20704439/103282748-08d67200-49d7-11eb-8a40-0a22156fdea5.jpg). As iOS knows which Apps are using the ENF it should be possible for Apple to add another sentence to this notification like e.g. "Deletion of this App will also delete all collected keys. They can not be recovered, not even through a cloud backup". I'm aware that the CWA-developers can't change underlying iOS-notifications but maybe there is a chance to voice this issue towards Apple.

However it was never only about the uninstallation of CWA but of the backup-and-restore process as @TR0LLIP0P pointed out in their initial post:

> So they will transfer their personal data (google and apple make this possible with cloud backup) and use the new phone, not the old one.
I suspect that most of them do not know that the collected IDs of CWA will NOT be transferred.

And this is what already became a problem way before Christmas, as I pointed out above:

> I would strongly encourage the developers to think about a one-time push notification explaining what @TR0LLIP0P pointed out and linking that notification to either the FAQ-page or an in-app explanation. I'm personally informed about two people who switched phones and haven't thought about this over the past weeks, I believe this might become a huge problem during Christmas.

This was my comment 20 days ago and now that Christmas eve has passed I already heard of another two people who switched phones while selling their old ones. They now have no possibility to get notified about past encounters as they both believed that the keys would be included in the iOS-backup - just like anything else is. The average user cannot be expected to know all the technical details of what's going on in the background, therefore I'm a bit upset that the chance to prevent events like this has passed. 

Still I would leave the issue open because people switching phones and not knowing about how the backups work is not something that will go away anytime soon.

**Comment by TR0LLIP0P** [comment]
I'm a little bit disappointed. Yesterday I got a notification that I should open the app (or something like that. The text was too long for the notification bar and after clicking on it, the app opened, but there were no infos). But this proves that it was (easily) possible to send a notification like "If you get a new phone, please be aware ... Thank you for using this app. Have a nice holiday" to the users before christmas.

@achisto knows at least 4 people whose IDs are lost because they didn't know about this problem.

@Ein-Tim 
> But the question is: When should the user get informed that the data is not included in the backup? At the frost start of the App? At a random time?
> At the first start of the App seems best, but I think normal users will quickly forget about this...

My idea was to send the notification a few days before christmas to all users, as stated in my post.
Because it's too late for that, it could be integrated in the screen when you open the app first:
You install apps on your new phone and open the cwa for the first time. A screen asks you "Is this the first time you downloaded the app or did you use it on another phone before?" If the user clicks "I used it before" (there has to be a choice, otherwise many user will not read the text!) there could be a text like "please be aware that your collected IDs were not transferred to this phone. For more information...."
